src/browserbox-imap.js

```diff
@@ -111,12 +117,7 @@
      */
     ImapClient.prototype.connect = function() {
         return new Promise((resolve, reject) => {
-            this.socket = this._TCPSocket.open(this.host, this.port, {
-                binaryType: 'arraybuffer',
-                useSecureTransport: this.secureMode,
-                ca: this.options.ca,
-                tlsWorkerPath: this.options.tlsWorkerPath
-            });
+            this.socket = this._TCPSocket.open(this.host, this.port, this.options.tcpSocket);
 
```

Comment:
Exactly. tcp-socket can be updated independently down the road. How it
handles its options object is its own responsibility. In general I think
it's good to minimize what BrowserBox need to know about the TCPSocket
object and it's shims.

In a truly ideal world I'd inject a configured TCPSocket into BrowserBox,
but that is a more invasive architectural change than I'm willing to commit
my efforts to. This covers most of the cases I can envision even if its a
pattern that falls short of true dependency inversion.


On Thu, Dec 10, 2015 at 1:15 PM, Felix Hammerl <notifications@github.com>
wrote:

> In src/browserbox-imap.js
> <https://github.com/whiteout-io/browserbox/pull/74#discussion_r47262340>:
>
> > @@ -111,12 +117,7 @@
> >       */
> >      ImapClient.prototype.connect = function() {
> >          return new Promise((resolve, reject) => {
> > -            this.socket = this._TCPSocket.open(this.host, this.port, {
> > -                binaryType: 'arraybuffer',
> > -                useSecureTransport: this.secureMode,
> > -                ca: this.options.ca,
> > -                tlsWorkerPath: this.options.tlsWorkerPath
> > -            });
> > +            this.socket = this._TCPSocket.open(this.host, this.port, this.options.tcpSocket);
> >
>
> true that. then we should pass in all the standardized socket options
> <https://developer.mozilla.org/en-US/docs/Web/API/TCPSocket> via the
> first options object, and the non-standard options for the shim in a second
> object (ca, ws, worker paths, ...). this would also require a change in the
> tcp-socket API, but that's a clean way to handle it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/whiteout-io/browserbox/pull/74/files#r47262340>.
>



-- 
Darrel O'Pry
The Spry Group, LLC.
http://www.spry-group.com
718-355-9767 x101